src/main/java/edu/kis/vh/nursery/list/IntLinkedList.java

```diff
@@ -1,40 +1,81 @@
 package edu.kis.vh.nursery.list;
 
+class Node {
+
+    private int value;
+    private Node prev, next;
+
+    public Node(int i) {
+        setValue(i);
+    }
+
+    public int getValue() {
+        return value;
+    }
+
+    public void setValue(int value) {
+        this.value = value;
+    }
+
+    public Node getPrev() {
+        return prev;
+    }
+
+    public void setPrev(Node prev) {
+        this.prev = prev;
+    }
+
+    public Node getNext() {
+        return next;
+    }
+
+    public void setNext(Node next) {
+        this.next = next;
+    }
+}
+
 public class IntLinkedList {
 
-	Node last;
-	int i;
-
-	public void push(int i) {
-		if (last == null)
-			last = new Node(i);
-		else {
-			last.next = new Node(i);
-			last.next.prev = last;
-			last = last.next;
-		}
-	}
-
-	public boolean isEmpty() {
-		return last == null;
-	}
-
-	public boolean isFull() {
-		return false;
-	}
-
-	public int top() {
-		if (isEmpty())
-			return -1;
-		return last.value;
-	}
-
-	public int pop() {
-		if (isEmpty())
-			return -1;
-		int ret = last.value;
-		last = last.prev;
-		return ret;
-	}
+    private static final int NO_ANY_ELEMENTS = -1;
+    private Node last;
+
+    public void push(int i) {
+        if (getLast() == null)
+            setLast(new Node(i));
+        else {
+            getLast().setNext(new Node(i));
+            getLast().getNext().setPrev(getLast());
+            setLast(getLast().getNext());
+        }
+    }
+
+    public boolean isEmpty() {
+        return getLast() == null;
+    }
+
+    public boolean isFull() {
+        return false;
+    }
+
+    public int top() {
+        if (isEmpty())
+            return NO_ANY_ELEMENTS;
+        return getLast().getValue();
+    }
+
+    public int pop() {
+        if (isEmpty())
+            return NO_ANY_ELEMENTS;
+        int ret = getLast().getValue();
+        setLast(getLast().getPrev());
+        return ret;
+    }
+
+    public Node getLast() {

```

Comment:
Shouldn't be encapsulated (typical private field). This is an exaggeration. Using private fields internally usually do not require encapsulation.